components/brave_wallet/browser/brave_wallet_service.h

```diff
@@ -6,39 +6,90 @@
 #ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_BRAVE_WALLET_SERVICE_H_
 #define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_BRAVE_WALLET_SERVICE_H_
 
+#include <list>
+#include <map>
 #include <memory>
+#include <string>
+#include <vector>
 
+#include "base/callback_forward.h"
+#include "base/containers/queue.h"
 #include "base/files/file_path.h"
 #include "base/macros.h"
 #include "base/memory/scoped_refptr.h"
 #include "base/memory/weak_ptr.h"
+#include "base/observer_list.h"
 #include "base/scoped_observer.h"
+#include "base/values.h"
 #include "components/keyed_service/core/keyed_service.h"
+#include "extensions/browser/extension_registry.h"
+#include "extensions/browser/extension_registry_observer.h"
+#include "url/gurl.h"
+
+class BrowserWalletDelegate;
+class PrefChangeRegistrar;
+class PrefService;
 
 namespace base {
+class FilePath;
 class SequencedTaskRunner;
 }  // namespace base
 
 namespace content {
 class BrowserContext;
 }  // namespace content
 
-class BraveWalletController;
-class Profile;
-
-class BraveWalletService : public KeyedService {
+class BraveWalletService : public KeyedService,
+    public extensions::ExtensionRegistryObserver {
  public:
-  explicit BraveWalletService(content::BrowserContext* context);
+  explicit BraveWalletService(content::BrowserContext* context,
+      std::unique_ptr<BrowserWalletDelegate> browser_wallet_delegate);
   ~BraveWalletService() override;
 
   bool Init();
-  BraveWalletController* controller() const { return controller_.get(); }
+
+  void ResetCryptoWallets(const base::FilePath& profile_path);
+  void CloseTabsAndRestart();
```

Comment:
shouldn't most of these be private?

Reply:
we'd have to add some friends to various API classes that way.